### DISCUSSION TITLE
Embed Tests into actual application project

### DISCUSSION THREAD

**Discussion by badcel** [thomhurst/TUnit#1169] [Ideas]
I wonder if it is possible to embed tests inside the project they are testing. This obviously not common in C# but I read about it in a Java context.

No more scrolling and searching for tests, they are where the code is. I think this is a nice feature. 

I understood from the readme, that the tests can run as a standalone binary. This is why I wondered if TUnit tests could be inside for example a regular console app.

**Comment by thomhurst** [comment]
Technically possible but I wouldn't recommend it to be honest.

TUnit creates a main method for you automatically. You'd need to create a manual entry point to run the tests.

**Comment by caio-bulgarelli** [comment]
@thomhurst, there is no need to create your own entrypoint. You just need to remove the normal project entrypoint when testing and let TUnit do it's magic.

Just add this to your csproj.

	
	<ItemGroup Condition="'$(Configuration)' == 'Release'">
		<Compile Remove="**\*.Tests.cs" />
	</ItemGroup>

	
	<ItemGroup Condition="'$(IsTestProject)' == 'true'">
		<Compile Remove="**\Program.cs" />
	</ItemGroup>

	
	<ItemGroup Condition="'$(Configuration)' != 'Release'">
		<PackageReference Include="TUnit" Version="1.2.3" />
		
	</ItemGroup>
